elastic/kibana
# [Console] Replace global `GET /_mapping` request with `GET <index>/_mapping`

## Summary

### Notes for reviewers 

- Currently, autocomplete suggestions for fields don't work with wildcards and data streams due to the [bug](https://github.com/elastic/kibana/issues/149496) in the `main`. It should be addressed separately. 

### How to test

In order to spot the loading behaviour, ideally you should create an index with a heavy mappings definition.  
Afterwards, write a query that requires a field from this index, e.g.:

```
GET <my-index>/_search
{
  "aggs": {
    "my_agg": {
      "terms": {
        "field": "",
        "size": 10
      }
    }
  }
}
```

Place a cursor next to the `field` property, it should trigger mappings fetch. After that, the mappings definition for this index will be cached and accessed synchronously. 

You can also open the browser's dev tools and enable Network throttling. It allows noticing loading behaviour for any index. 

--------------------

Resolves https://github.com/elastic/kibana/issues/146855

Instead of fetching all mappings upfront, requests mapping definition on demand per index according to the cursor position. 

Considering there is a maximum response size limit of 10MB in the `/autocomplete_entities` endpoint, field autocompletion wasn't working at all if the overall mappings definition exceeded this size.  Retrieving mappings per index tackles this and improves the init time. 

![Jan-25-2023 17-16-31](https://user-images.githubusercontent.com/5236598/214616790-4954d005-e56f-49f9-be6d-435c076270a8.gif)

### Checklist
